src/symbolic_regression/symbolic_regression.jl

```diff
@@ -0,0 +1,125 @@
+using SymbolicRegression
+using DataDrivenDiffEq
+
+
+"""
+$(TYPEDEF)
+
+Options for using SymbolicRegression.jl within the `solve` function.
+Automatically creates [`Options`](https://astroautomata.com/SymbolicRegression.jl/stable/api/#Options) with the given specification.
+Sorts the operators stored in `functions` into unary and binary operators on conversion.
+
+# Fields
+$(FIELDS)
+
+"""
+struct EQSearch <: AbstractSymbolicRegression
+    "Operators used for symbolic regression"
+    functions::AbstractVector{Function}
+    "Additionally keyworded arguments passed to SymbolicRegression.Options"
+    kwargs
+end
+
+function EQSearch(functions = Function[/, +, *, exp, cos]; kwargs...)
+    return EQSearch(functions, kwargs)
+end
+
+function to_options(x::EQSearch)
+    binops = Tuple([fi for fi in x.functions if is_binary(fi)])
+    unaops = Tuple([fi for fi in x.functions if is_unary(fi)])
```

Comment:
How does this work if one passes `-` as an operator? Since it can be both negative, or subtraction. Does it get replicated in both operator lists?

Reply:
Also - I think "ariety" should be "arity", right? https://en.wikipedia.org/wiki/Arity

Reply:
(or more generally, if a user defines a function that can take either 1 or 2 arguments?) Maybe it would be good to allow explicit declaration of arity, at least for the SymbolicRegression options?

Reply:
> How does this work if one passes `-` as an operator? Since it can be both negative, or subtraction. Does it get replicated in both operator lists?

Hm, interesting. I would assume that "-1" is a number and not an operator and a number. Do you use it also as a unary operator?

Reply:
> Also - I think "ariety" should be "arity", right? https://en.wikipedia.org/wiki/Arity

I also have problems with german word spelling. This seems fair :smile:

Reply:
> Hm, interesting. I would assume that "-1" is a number and not an operator and a number. Do you use it also as a unary operator?

I don't usually use the negation operator, but it is a possibility. e.g., `neg(x)` has complexity of 2, while `-1.0 * x` has complexity 3, and also might complicate things since it is another constant to optimize. e.g., `Options(..., unary_operators=(-,), binary_operators=(-,))` would allow for use of the operator in both negation and subtraction.

Reply:
I think this could be done rather nicely.  And I found a bug.

```julia
function check_ariety(f::Function, n::Int,  t::Type = Number)
           for m in methods(f, Tuple([t for i in 1:n]))
               m.nargs-1 == n && return true
           end
           return false
end
```

Which returns `true` if the function `f` is defined for `n` arguments over type(s) `t`.